test_conformance/allocations/main.cpp

```diff
@@ -416,8 +397,8 @@ void printUsage(const char *execName)
              "of the memory objects.\n");
     log_info("\n");
     log_info("Test names (Allocation Types):\n");
-    for (int i = 0; i < test_num; i++)
+    for (int i = 0; i < test_registry::getInstance().num_tests(); i++)
     {
-        log_info("\t%s\n", test_list[i].name);
+        log_info("\t%s\n", test_registry::getInstance().definitions()[i].name);
     }
```

Comment:
Doesn't this code duplicate the corresponding feature in the harness? Can we get rid of it altogether?

Reply:
It does, but this test suite has a custom arguments parsing logic, and a custom help message, `printUsage`. Removing that still prints the general usage help, but does not list additional options such as `randomize`, `single`, `multiple`, etc.